Added single module occupancy plots for on track digis and charge Vs …

akhatiwa opened the pull request:
…size plots (on track) per layer/disk

akhatiwa committed:
Added single module occupancy plots for on track digis and charge Vs size plots (on track) per layer/disk

cmsbuild commented:
@akhatiwa, CMSSW_9_4_X branch is closed for direct updates. cms-bot is going to move this PR to master branch.
In future, please use cmssw master branch to submit your changes.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20567/784

cmsbuild commented:
A new Pull Request was created by  @akhatiwa  for master.

It involves the following packages:

DQM/SiPixelPhase1TrackClusters


@kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please review it and eventually sign? Thanks.
@idebruyn, @threus, @fioriNTU, @hdelanno this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: a784c3dfad8fa3a984203208f32c509467176989
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23067/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/e133687c2dd23771612485bf4baf046b9317e8e8
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23067/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23067/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20567/859

cmsbuild commented:
Pull request #20567 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

fioriNTU commented on DQM/SiPixelPhase1TrackClusters/src/SiPixelPhase1TrackClusters.cc:
```diff
@@ -211,10 +223,15 @@ void SiPixelPhase1TrackClusters::analyze(const edm::Event& iEvent, const edm::Ev
        histo[SiPixelPhase1TrackClustersOnTrackShape].fill(shape?1:0,id, &iEvent);
       }
 
+      std::cout<<"Cluster charge: "<<charge<<endl;
```
@akhatiwa please remove this cout

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20567/860

cmsbuild commented:
Pull request #20567 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

fioriNTU commented on DQM/SiPixelPhase1TrackClusters/src/SiPixelPhase1TrackClusters.cc:
```diff
@@ -211,10 +223,15 @@ void SiPixelPhase1TrackClusters::analyze(const edm::Event& iEvent, const edm::Ev
+          histo[SiPixelPhase1PixelChargeOnTrack].fill(vecipxl.adc, id, &iEvent);
       }
+       
+      histo[SiPixelPhase1NPixelsByNDigisOnTrack].fill(cluster.size()/input.product()->size(), id, &iEvent);
```
this is not the Histogram I asked for, you should count ALL the digis (i.e the Pixels) that belongs to on-track clusters and at the end of the loop on tracks plot the ratio between the total number of on-track digis in the event and the total number of PixelDigis

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20567/861

cmsbuild commented:
Pull request #20567 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

boudoul commented:
Adding @gflouris for information

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: 065ba72bd1a2f0fbd775194e6299c49d25c13a7e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23228/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/6d77b6936b3ea25f71264cc8bc6bf50d7ca45f67
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23228/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23228/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23228/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2691683
 - DQMHistoTests: Total failures: 211
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2691283
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 15 edm output root files, 26 DQM output files

davidlange6 commented on DQM/SiPixelPhase1TrackClusters/src/SiPixelPhase1TrackClusters.cc:
```diff
@@ -209,13 +211,20 @@ void SiPixelPhase1TrackClusters::analyze(const edm::Event& iEvent, const edm::Ev
       }
 
+      for (int i =0; i<cluster.size(); i++){
+          SiPixelCluster::Pixel vecipxl = cluster.pixel(i);
```
hi @akhatiwa - can this be a const & SiPixelCluster::Pixel instead so to avoid the creation of a copy here?

akhatiwa replied:
@davidlange6 - that's probably better.

davidlange6 commented:
hi @akhatiwa - beyond my original comment, could you also estimate how many histogram bins are added by this PR? Thanks

akhatiwa committed:
Update SiPixelPhase1TrackClusters.cc

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20567/1042

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20567/1042/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20567/1042/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly

dmitrijus commented:
please apply the code checks

akhatiwa commented:
@dmitrijus, I'm not sure why the code checks is being triggered. "const & SiPixelCluster::Pixel; vecipxl = cluster.pixel(i);" doesn't make sense.

davidlange6 commented:
I agree - @smuzaffar have we seen something like this before?
https://cmssdt.cern.ch/SDT/code-checks/PR-20567/1042/git-diff.patch

> On Oct 3, 2017, at 7:49 PM, akhatiwa <notifications@github.com> wrote:
> 
> @dmitrijus, I'm not sure why the code checks is being triggered. "const & SiPixelCluster::Pixel; vecipxl = cluster.pixel(i);" doesn't make sense.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

fioriNTU commented:
@davidlange6 for the number of bins, consider that the module occupancy plots (one bin per pixel) will be produced only in the Online system, this is the "big load". I have made some bin optimization lately (not yet in production) that should be more than enough to compensate this addition. @akhatiwa please provide David the total number of bins you added, separating the single module occupancy by the rest. You can even save a factor 2 on the bin number setting the binning of the cluster size to 15 (instead of 30).

akhatiwa commented:
@dmitrijus, @davidlange6: I guess it should be SiPixelCluster::Pixel const & vecipxl = cluster.pixel(i); instead?

akhatiwa committed:
Update SiPixelPhase1TrackClusters.cc

Modified "const & SiPixelCluster::Pixel vecipxl = cluster.pixel(i);" to "SiPixelCluster::Pixel const & vecipxl = cluster.pixel(i);"

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20567/1124

cmsbuild commented:
Pull request #20567 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

akhatiwa committed:
Update SiPixelPhase1TrackClusters_cfi.py

Modified range_nbins for SiPixelPhase1ClustersChargeVsSizeOnTrack from 30 to 15, range_y_max from 150e3 to 80e3, and range_y_nbins from 150 to 100.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #20567 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

akhatiwa commented:
@davidlange6, The x-y nbins for SiPixelPhase1DigisHitmapOnTrack is 416 X 160 produced per module (online only) as mentioned by @fioriNTU above. 
For SiPixelPhase1ClustersChargeVsSizeOnTrack the x-y nbins is 15 X 100 produced per Barrel Layer/FPix Disk.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: d2e63c1ffc6aa994a1adaedc8d5d30853dba73bd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23428/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/575a47c1dc033097d7d5fbda7e2fe14f71ee98b9
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23428/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23428/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20567/23428/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2761357
 - DQMHistoTests: Total failures: 251
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2760919
 - DQMHistoTests: Total skipped: 187
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 10 edm output root files, 26 DQM output files

davidlange6 commented:
+1
